src/controller/handler/connections.ts

```diff
@@ -37,38 +36,75 @@ export class Connections implements IAgentResponseHandler {
+            this.agentGovernance.readPermission('connections', governanceKey);
+            this.cache.set(cacheKey, {});
+        }
+
         if (route !== 'topic' || topic !== 'connections') {
             throw new ProtocolException('Connections',`${route}/${topic} is not valid.`);
         }
 
-        if (body.state !== 'request') {
-            Logger.info(`${body.state} not applicable`);
-            return;
+        const governanceKey = `${body.state}-${body.initiator}`;
+        const cacheKey = `${agentId}-${governanceKey}`;
+        const permissionState = this.agentGovernance.peekPermission('connections', governanceKey);
+
+        if (AgentGovernance.PERMISSION_DENY === permissionState) {
+             throw new ProtocolException('AgencyGovernance',`${governanceKey} governance doesnt not allow.`);
         }
 
-        if (AgentGovernance.PERMISSION_DENY === this.agentGovernance.getPermission("connections", "accept-invitation")) {
-            throw new ProtocolException('AgencyGovernance',`${topic} governance doesnt not allow.`);
+        // if the cacheKey is in the cache then the agent has already accepted the request
+        // when we only allow once, there is no need to continue with this message
+        if (this.cache.get<any>(cacheKey) && permissionState === AgentGovernance.PERMISSION_ONCE) {
+             return;
         }
 
-        // if the agentUrl is in the cache then the agent has already accepted the request
-        if (this.cache.get<any>(agentUrl)) {
-            return;
+        // this webhook message indicates an agent received an connection
+        // invitation and we want to tell them to accept it, if the policy allows
+        if (body.state === 'invitation' && body.initiator === 'external') {
+            readPermission(governanceKey, cacheKey);
+            await delay(2000);
+
+            let url: string = agentUrl + `/${Connections.CONNECTIONS_URL}/${body.connection_id}/accept-invitation`;
+            const req: AxiosRequestConfig = {
+                method: 'POST',
+                url,
+                headers: {
+                    'x-api-key': adminApiKey,
+                }
+            };
+
+            Logger.info(`requesting agent to accept connection invite ${req.url}`);
+            const res = await this.http.requestWithRetry(req);
+            return res;
         }
 
-        await this.cache.set(agentUrl, {  }, { ttl: 0});
-        let url: string = agentUrl + `/${Connections.CONNECTIONS_URL}/${body.connection_id}/accept-request`;
+        // this webhook message indicates the receiving agent has accepted the invite and now
+        // we need to instruct this agent to finish the steps of a connection
+        if (body.state === 'request' && body.initiator === 'self') {
+            readPermission(governanceKey, cacheKey);
+            await delay(2000);
+
+            let url: string = agentUrl + `/${Connections.CONNECTIONS_URL}/${body.connection_id}/accept-request`;
```

Comment:
I wonder if we should have one json entry for "accept-invites" and one for "accept-request" so that these can be controlled independently. It's not super important right now but would allow more fine grained control.